decorator/UnicodeParsed.java

```diff
@@ -0,0 +1,40 @@
+package decorator;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import model.ParsedInterface;
+
+public class UnicodeParsed implements ParsedInterface {
+
+  private final ParsedInterface origin;
+
+  UnicodeParsed(ParsedInterface parser) {
+
+    this.origin = parser;
+  }
+
+  @Override
+  public File giveFile() {
+    return this.origin.giveFile();
+  }
+
+  @Override
+  public String giveContent() throws IOException {
```

Comment:
@undsett why not use origin `giveFile()` to get the content? Exactly this is called decoration, just lay on origin and do your job